File: js/tests/wordEndings.test.js
const addEnding = require("../wordEndings");

const data = [
  [
    ["clever", "meek", "hurried", "nice"],
    "ly",
    ["cleverly", "meekly", "hurriedly", "nicely"]
  ],
  [["new", "pander", "scoop"], "er", ["newer", "panderer", "scooper"]],
  [["bend", "sharpen", "mean"], "ing", ["bending", "sharpening", "meaning"]],
  [["bend", "tooth", "mint"], "y", ["bendy", "toothy", "minty"]],
  [["bend", "tooth", "mint"], "ier", ["bendier", "toothier", "mintier"]]
];

describe("addEnding", () => {
  describe("Add string ending to each member of an array", () => {
    test.each(data)("addEnding(%p, %p) = %p", (arr, ending, expected) => {
      var actual = addEnding(arr, ending);

      expect(actual).toEqual(expected);
    });
  });
});

File: CSharp/Tests/WordEndingsTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class WordEndingsTest
    {
        [Theory]
        [InlineData(new string[] { "clever", "meek", "hurried", "nice" }, "ly", new string[] { "cleverly", "meekly", "hurriedly", "nicely" })]
        [InlineData(new string[] { "new", "pander", "scoop" }, "er", new string[] { "newer", "panderer", "scooper" })]
        [InlineData(new string[] { "bend", "sharpen", "mean" }, "ing", new string[] { "bending", "sharpening", "meaning" })]
        [InlineData(new string[] { "bend", "tooth", "mint" }, "y", new string[] { "bendy", "toothy", "minty" })]
        [InlineData(new string[] { "bend", "tooth", "mint" }, "ier", new string[] { "bendier", "toothier", "mintier" })]
        public void AddEnding_StringArrayAndStringEnding_ReturnStringArrayWithEndingOnEachElement(string[] arr, string ending, string[] expected)
        {
            var actual = WordEndings.AddEnding(arr, ending);

            Assert.Equal(expected, actual);
        }
    }
}